software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/MongoDBClientSupport.java

```diff
@@ -96,12 +111,18 @@ private HostAndPort getServerHostAndPort() {
         MongoClient client = client();
         try {
             DB db = client.getDB(database);
-            CommandResult status;
-            try {
-                status = db.command(command);
-            } catch (MongoException e) {
-                LOG.warn("Command " + command + " on " + getServerAddress() + " failed", e);
-                return Optional.absent();
+            CommandResult status = null;
+            // The mongoDB client can occasionally fail to connect. Try up to 5 times to run the command
+            for (int i = 0; i < 5; i++) {
```

Comment:
Would a `Repeater` be of any use? It allows you to give backoffs, etc. It's a cleaner way of expressing this block.